velox/substrait/SubstraitParser.cpp

```diff
@@ -60,6 +60,14 @@ std::shared_ptr<SubstraitParser::SubstraitType> SubstraitParser::parseType(
       nullability = substraitType.string().nullability();
       break;
     }
+    case ::substrait::Type::KindCase::kUserDefined: {
+      // We only support UNKNOWN type to handle the null literal whose type is
```

Comment:
This logic is coupled with the logic in checkTypeExtension. It would be nice to refactor to consolidate all logic for handing user-defined types in a single place. This can be done in a follow-up PR though.